cordformation/src/noderunner/kotlin/net/corda/plugins/NodeRunner.kt

```diff
@@ -124,52 +123,59 @@ private abstract class JavaCommand(
 }
 
 private class HeadlessJavaCommand(jarType: JarType, dir: File, debugPort: Int?, monitoringPort: Int?, args: List<String>, jvmArgs: List<String>)
-    : JavaCommand(jarType.jarName, dir, debugPort, monitoringPort, jarType.enableJolokia, dir.name, jarType.headlessArgs + args, jvmArgs) {
+    : JavaCommand(jarType.jarName, dir, debugPort, monitoringPort, jarType.enableJolokia, dir.name, if(isTmux()) args else jarType.headlessArgs + args, jvmArgs) {
```

Comment:
This check can be replaced entirely by modifying line 31 from

val headless = GraphicsEnvironment.isHeadless() || args.contains(HEADLESS_FLAG)

to

     val headless = ((!isTmux() && GraphicsEnvironment.isHeadless()) || args.contains(HEADLESS_FLAG))

That way the any tmux terminal isn't automatically set to headless and will thus behave normally. This will also mean the headless flag still works as intended (by disabling the shell).

Similar the conditional on line 129 can be reverted.

Reply:
yeah then I won't even need any of this  function `getCommandAsPerOS` actually just a ` val headless = ((!isTmux() && GraphicsEnvironment.isHeadless()) || args.contains(HEADLESS_FLAG))` will do the trick,

Reply:
If I just undo everything and make ` val headless = ((!isTmux() && GraphicsEnvironment.isHeadless()) || args.contains(HEADLESS_FLAG))` it does solve the problem and everything works fine, I don't think there is a need to abstract and create this function as well `getCommandAsPerOS` what do you think?